app/services/event_builders/review.rb

```diff
@@ -5,13 +5,22 @@ class Review < EventBuilder
     def build
       review_data = @payload['review']
       Events::Review.find_or_create_by!(github_id: review_data['id']) do |review|
-        review.owner = find_or_create_user(review_data['user'])
-        review.pull_request = find_pull_request
+        assign_attrs(review, review_data)
 
         ATTR_PAYLOAD_MAP.each do |key, value|
           review.public_send("#{key}=", review_data.fetch(value))
         end
       end
     end
+
+    def assign_attrs(review, review_data)
+      review.owner = find_or_create_user(review_data['user'])
+      review.pull_request = find_pull_request
+
+      owner = review.owner
+      pull_request = review.pull_request
+      review.review_request = find_or_create_review_request(pull_request, owner.id)
+      owner.projects << pull_request.project
```

Comment:
question, the code above is "assigning" but not persisting. But this line is actually persisting the change, can it have any negative impact like leaving system in an invalid state if something fails?

Reply:
I got an error from reek that I was doing `review.owner` and `review.pull_request` twice, so assigning solve that. I will add an exception on reek i guess